### PR TITLE
Support Swift3 (CocoaPods, Carthage)

### PR DESCRIPTION
**Pull request by muukii**
- [-] Build success (iOS, tvOS, watchOS, macOS)
- [x] Tests (Quick, Nimble is not avaliable in Swift3)
- [x] Travis CI
- [x] `pod lib lint`  okay but, tvOS is invalid ->  https://github.com/CocoaPods/CocoaPods/issues/5320
- [ ] Push CocoaPods 

**notes**

macOS support version 10.9 -> 10.10 (caused by RxSwift 3.0)

Related : https://github.com/RxSwiftCommunity/RxOptional/issues/25

### CODE REVIEW

**Comment by ashfurrow** [comment]
Quick + Nimble should have Swift 3 support. The branches they're using are listed here: https://github.com/Moya/Moya/pull/608/files#diff-c2c2f4d15bfae1123d3167a807ac48d6R18

**Comment by ashfurrow** [comment]
And thanks for the PR! Looks great so far.

**Comment by muukii** [comment]
@ashfurrow 
Thank you.
I will try it now. (Quick, Nimble)

**Comment by muukii** [comment]
@ashfurrow 

What is wrong?

screen shot 2016-09-20 at 12 28 09 am

**Comment by ashfurrow** [comment]
I've not seen that before, interesting! I would try introducing types, like:

```swift
subscribeNext { (value: [[Int]]) in
  expect(value).to( equal([[1], [3, 4], [5]]) )
}
```

And see if that works. If you can't get it, don't worry! Just leave a line note and I can take a look.

**Comment by muukii** [comment]
@ashfurrow 
TravisCI has been fixed!
The rest is the only modification of the test code!

**Comment by muukii** [comment]
@ashfurrow 
Can you take a look test-code?
https://github.com/RxSwiftCommunity/RxOptional/pull/28#issuecomment-248026750

**Comment by ashfurrow** [comment]
This is working on CI, is it good to merge?

**Comment by muukii** [comment]
@ashfurrow 
Is not solved this test-code. https://github.com/RxSwiftCommunity/RxOptional/pull/28#issuecomment-248026750
https://github.com/RxSwiftCommunity/RxOptional/pull/28/files#diff-dce1d7221f8d848bbfb8c34adcea6fc7R17

CI only have a build preparation. Not running test.

File: Test/OccupiableOperatorsTests.swift [hunk 1]
```diff
@@ -14,7 +14,9 @@ class OccupiableOperatorsSpec: QuickSpec {
                         .filterEmpty()
                         .toArray()
                         .subscribe(onNext: {
-                            expect($0).to(equal([[1], [3, 4], [5]]))
+                            expect($0[0]).to(equal([1]))
+                            expect($0[1]).to(equal([3, 4]))
+                            expect($0[2]).to(equal([5]))
```
Comment (muukii) [inline Test/OccupiableOperatorsTests.swift]: @ashfurrow  
Sorry.
The other in a good way did not come up with..

Comment (ashfurrow) [inline Test/OccupiableOperatorsTests.swift]: If this works, then let's use it :+1:

**Comment by ashfurrow** [comment]
I've restarted the failing build, I think it should be fine. @muukii is this pull request okay to merge? Or are there other changes you'd like to make first?

**Comment by muukii** [comment]
@ashfurrow 
> I think it should be fine.
I think so too. Let's merge.

I'll fix conflict

**Comment by muukii** [comment]
@ashfurrow 
I was rebase from master.
if CI is working, let's merge.

**Comment by ashfurrow** [comment]
Thanks a lot for contributing! I've invited you to join the RxSwiftCommunity GitHub organization – no pressure to accept! If you'd like more information on what that means, check out our [contributor guidelines](https://github.com/RxSwiftCommunity/contributors) and feel free to reach out with any questions.

**Comment by ashfurrow** [comment]
I'll do a release sometime today, thanks again!

**Comment by muukii** [comment]
@ashfurrow

Thank you!!!

Thank you for invite.
I've joined RxSwiftCommunity!

**Comment by ashfurrow** [comment]
I've pushed a 3.0.0 tag, @thanegill can you add me as an owner to the podspec so I can push it to CocoaPods trunk?

**Comment by thanegill** [comment]
@ashfurrow I add you under ash@ashfurrow.com.

**Comment by ashfurrow** [comment]
Okay, thanks! Pushed! https://cocoapods.org/pods/RxOptional

**Comment by he9lin** [comment]
Great work all. Just FYI, I just tried the new commit, and had to include `Quick` in my `Cartfile` to be able to use it. Otherwise, running `carthage update` would give me the following error:

```
RxOptional/Test/OccupiableOperatorsTests.swift:1:8: error: no such module 'Quick'
```

**Comment by ashfurrow** [comment]
I'm not sure how to fix that, can anyone else chime in? @he9lin if you know how to fix it, would you submit a PR? Thanks for letting us know!

**Comment by he9lin** [comment]
@ashfurrow I'll try dig into it more as I'm new to Carthage :)

**Pull request merged**